packages/react-components/src/components/SendBox.tsx

```diff
@@ -243,18 +214,22 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
     }
 
     // Don't send message until all files have been uploaded successfully
+    /* @conditional-compile-remove(file-sharing) */
     setFileUploadsPendingError(undefined);
 
-    if (hasIncompleteFileUploads(props)) {
-      /* @conditional-compile-remove(file-sharing) */
+    /* @conditional-compile-remove(file-sharing) */
+    if (hasIncompleteFileUploads(activeFileUploads)) {
       setFileUploadsPendingError({ message: strings.fileUploadsPendingError, timestamp: Date.now() });
       return;
     }
 
     const message = textValue;
     // we don't want to send empty messages including spaces, newlines, tabs
     // Message can be empty if there is a valid file upload
-    if (!EMPTY_MESSAGE_REGEX.test(message) || hasFile(props)) {
+    if (
+      !EMPTY_MESSAGE_REGEX.test(message) ||
+      /* @conditional-compile-remove(file-sharing) */ hasCompletedFileUploads(activeFileUploads)
```

Comment:
Have we made changes on this?

Reply:
Yes, it was part of the latest Patrick's PR

Reply:
https://github.com/Azure/communication-ui-library/pull/4034

Reply:
part of my refactor as well